### PR TITLE
upstream material-ui to v5

### PR DESCRIPTION
**Pull request by msivasubramaniaan**
Signed-off-by: msivasubramaniaan <msivasub@redhat.com>

Fix: https://github.com/redhat-developer/vscode-openshift-tools/issues/2689

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -119,15 +119,12 @@
 		"yaml": "^1.10.2"
 	},
 	"devDependencies": {
-		"@emotion/react": "^11.7.1",
-		"@emotion/styled": "^11.6.0",
-		"@material-ui/core": "^4.12.3",
```
Comment (mohitsuman) [inline package.json:124]: @msivasubramaniaan As we have removed this, should the reference of this in files be also updated such as here: https://github.com/redhat-developer/vscode-openshift-tools/blob/main/src/webview/cluster/app/cluster.style.tsx#L6

We need to be update all the location where the references are added for old dependency.

**Review by mohitsuman — CHANGES_REQUESTED** [review]

**Comment by msivasubramaniaan** [comment]
@mohitsuman as the reference of createStyles from v4 to v5 internally taken care through @mui/systems. Now the build has success

File: package.json [hunk 2]
```diff
@@ -119,15 +119,12 @@
...
-		"@material-ui/icons": "^4.11.3",
-		"@material-ui/lab": "^4.0.0-alpha.60",
+		"@emotion/react": "^11.10.5",
+		"@emotion/styled": "^11.10.5",
 		"@mui/core": "^5.0.0-alpha.54",
 		"@mui/icons-material": "^5.2.5",
 		"@mui/lab": "^5.0.0-alpha.61",
-		"@mui/material": "^5.4.1",
+		"@mui/material": "^5.10.16",
```
Comment (mohitsuman) [inline package.json:127]: Let's keep it to the latest version 5.11.0

File: package.json [hunk 3]
```diff
@@ -119,15 +119,12 @@
...
-		"@emotion/react": "^11.7.1",
-		"@emotion/styled": "^11.6.0",
-		"@material-ui/core": "^4.12.3",
-		"@material-ui/icons": "^4.11.3",
-		"@material-ui/lab": "^4.0.0-alpha.60",
+		"@emotion/react": "^11.10.5",
+		"@emotion/styled": "^11.10.5",
 		"@mui/core": "^5.0.0-alpha.54",
 		"@mui/icons-material": "^5.2.5",
 		"@mui/lab": "^5.0.0-alpha.61",
```
Comment (mohitsuman) [inline package.json:126]: We should also update the @mui/lab version to latest 5.0.0-alpha.112

File: package.json [hunk 4]
```diff
@@ -119,15 +119,12 @@
...
 	"devDependencies": {
-		"@emotion/react": "^11.7.1",
-		"@emotion/styled": "^11.6.0",
-		"@material-ui/core": "^4.12.3",
-		"@material-ui/icons": "^4.11.3",
-		"@material-ui/lab": "^4.0.0-alpha.60",
+		"@emotion/react": "^11.10.5",
+		"@emotion/styled": "^11.10.5",
 		"@mui/core": "^5.0.0-alpha.54",
 		"@mui/icons-material": "^5.2.5",
```
Comment (mohitsuman) [inline package.json:125]: Similarly with the icons-material to latest.

File: package.json [hunk 5]
```diff
@@ -119,15 +119,12 @@
...
 		"@mui/lab": "^5.0.0-alpha.61",
-		"@mui/material": "^5.4.1",
+		"@mui/material": "^5.10.16",
 		"@patternfly/react-core": "^4.198.19",
```
Comment (mohitsuman) [inline package.json:128]: @msivasubramaniaan are we using anything patternfly in our codebase ?

File: package.json [hunk 6]
```diff
@@ -119,15 +119,12 @@
...
 		"@rjsf/material-ui": "^3.2.1",
```
Comment (mohitsuman) [inline package.json:129]: Similarly are we using this module anywhere ? As this material UI still point to v4 and we have to move to v5. So need to check if this is needed.

**Review by mohitsuman — CHANGES_REQUESTED** [review]

**Review by mohitsuman — APPROVED** [review]

**Pull request merged**